lua/smh/modifiers/advlights.lua

```diff
@@ -0,0 +1,123 @@
+
+MOD.Name = "Soft Lamps";
+
+function MOD:IsAdvLight(entity)
+	
+	local theclass = entity:GetClass();
+	
+	if theclass ~= "projected_light" and theclass ~= "projected_light_new" and theclass ~= "cheap_light" and theclass ~= "expensive_light" and theclass ~= "expensive_light_new" and theclass ~= "spot_light" then return false; end -- probably there must be a better way to do this. all advanced light entities share "base_light" as baseclass, just in case
```

Comment:
You can at least avoid the long if clause by putting all the class names into a table.

```
local validClasses = {
    "projected_light" = true,
    "projected_light_new" = true,
    ...
};
return validClasses[theclass] or false;
```